diff --git a/build.gradle b/build.gradle
index 30bac4265a..217f86dd36 100644
--- a/build.gradle
+++ b/build.gradle
@@ -308,7 +308,8 @@ if (IS_WINDOWS && OS_ARCH != "x86" && OS_ARCH != "amd64") {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
 } else if (IS_MAC && OS_ARCH != "x86_64" && OS_ARCH != "aarch64") {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
-} else if (IS_LINUX && OS_ARCH != "i386" && OS_ARCH != "amd64" && OS_ARCH != "aarch64") {
+} else if (IS_LINUX && OS_ARCH != "i386" && OS_ARCH != "amd64" && OS_ARCH != "aarch64"
+        && OS_ARCH != "arm" && OS_ARCH != "ppc64le" && OS_ARCH != "s390x") {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
 }
 
