diff --git a/buildSrc/linux.gradle b/buildSrc/linux.gradle
index 10f595a4c7..f6642edf17 100644
--- a/buildSrc/linux.gradle
+++ b/buildSrc/linux.gradle
@@ -45,7 +45,7 @@ def commonFlags = [
         "-fstack-protector",
         "-Wextra", "-Wall", "-Wformat-security", "-Wno-unused", "-Wno-parentheses", "-Werror=trampolines"] // warning flags
 
-if (!IS_64) {
+if (OS_ARCH == "i386") {
     commonFlags += "-m32"
 }
 
